hack/hcp-sre-breakglass.sh

```diff
@@ -0,0 +1,128 @@
+# In case of errors, resources may be left behind, etc.
+# TLDR: this script is the worst but good enough for a quick and dirty POC.
+#
+# This is an experiment script that creates a kubeconfig for an HCP using the sre-break-glass signer.
+# This script is intended to be run towards the management cluster that hosts the HCP we want to access.
+#
+# The resulting kubeconfig will grant membership in the sre-group group within the HCP.
+# An ACM policy grants that group access to the sre-role, which is a simple dummy role that allows listing all namespace.
+#
+# Example usage:
+# ./hcp-sre-breakglass.sh <cs-cluster-id>
+#
+
+SUBJECT="/CN=system:sre-break-glass:${USER}/O=sre-group"
+
+CS_CLUSTER_ID=$1
+HOSTED_CLUSTER_CR=$(kubectl get hostedcluster -A -l "api.openshift.com/id=${CS_CLUSTER_ID}" -o yaml | yq -r '.items[0]')
+HCP_NAMESPACE=$(echo "${HOSTED_CLUSTER_CR}" | yq '.metadata.namespace + "-" + (.metadata.labels["api.openshift.com/name"] | tostring)')
+HCP_API_SERVER=$(echo "${HOSTED_CLUSTER_CR}" | yq '.status.controlPlaneEndpoint.host')
+
+# download the API server CA from the HCP_API_SERVER_URL
+echo "Downloading the API server CA from ${HCP_API_SERVER}"
+HCP_ROOT_CA_CRT=$(openssl s_client -servername "${HCP_API_SERVER}" -connect "${HCP_API_SERVER}:443" </dev/null 2>/dev/null | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' | base64 | tr -d '\n')
+
+KEY_FILE="sre-breakglass-${CS_CLUSTER_ID}-${USER}.key"
+CSR_NAME="sre-breakglass-${CS_CLUSTER_ID}-${USER}"
+CSR_FILE="${CSR_NAME}.csr"
+KUBECONFIG_FILE="${CSR_NAME}.kubeconfig"
+
+# Generate a private key
+echo "Create private key"
+openssl genrsa -out "${KEY_FILE}" 2048
+
+# Generate a CSR using the private key and subject
+echo "Create CSR with subject ${SUBJECT}"
+openssl req -new -key "${KEY_FILE}" -subj "${SUBJECT}" -out "${CSR_FILE}"
+
+# Base64 encode the CSR
+CSR_BASE64=$(base64 -i "${CSR_FILE}" | tr -d '\n')
+
+# Create a Kubernetes CSR manifest
+cat <<EOF | kubectl apply -f -
+apiVersion: certificates.k8s.io/v1
+kind: CertificateSigningRequest
+metadata:
+  name: ${CSR_NAME}
+  labels:
+    api.openshift.com/id: $CS_CLUSTER_ID
+    api.openshift.com/name: $USER
+    api.openshift.com/type: break-glass-credential
+spec:
+  expirationSeconds: 86353
+  request: ${CSR_BASE64}
+  signerName: hypershift.openshift.io/${HCP_NAMESPACE}.sre-break-glass
+  usages:
+  - client auth
+  - digital signature
+EOF
+
+echo "Kubernetes CSR ${CSR_NAME} applied to the cluster."
+
+cat <<EOF | kubectl apply -f -
+apiVersion: certificates.hypershift.openshift.io/v1alpha1
+kind: CertificateSigningRequestApproval
+metadata:
+  creationTimestamp: null
+  labels:
+    api.openshift.com/id: $CS_CLUSTER_ID
+    api.openshift.com/name: $USER
+    api.openshift.com/type: break-glass-credential
+  name: ${CSR_NAME}
+  namespace: ${HCP_NAMESPACE}
+EOF
+
+echo "Hypershift CSR Approval ${CSR_NAME} applied to the cluster."
+
+# wait for the CSR to be approved
+while true; do
```

Comment:
`kubectl wait --for` on conditions is what you want here :)

Reply:
good point. done